### PR TITLE
DPP-253 PoC Design Non-Consuming Choice Subscriptions

### PR DESCRIPTION
**Pull request by meiersi-da**
This PR is not meant to be merged. I've created this Draft PR so I could easily provide feedback on the changes to @mzagorski-da and @kamil-da . 

I expect either of the two to open a proper PR in due time.

### CODE REVIEW

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/transaction.proto [hunk 1]
```diff
@@ -72,7 +72,7 @@ message TransactionTree {
 message TreeEvent {
     oneof kind {
         CreatedEvent created = 1;
-        ExercisedEvent exercised = 2;
+        ExercisedEvent exercised = 2; // should be introduced into transactions without children; needs modification
```
Comment (meiersi-da) [inline ledger-api/grpc-definitions/com/daml/ledger/api/v1/transaction.proto:75]: @mzagorski-da : what kind of modification are you thinking about? Can you change the PR to contain it, so we can align on it?

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/event.proto [hunk 2]
```diff
@@ -25,11 +25,12 @@ option csharp_namespace = "Com.Daml.Ledger.Api.V1";
 message Event {
   oneof event {
     CreatedEvent created = 1;
+    ExercisedEvent exercised = 2;
     ArchivedEvent archived = 3;
  }
-  // see https://github.com/digital-asset/daml/issues/960
-  reserved 2;
-  reserved "exercised";
+//  // see https://github.com/digital-asset/daml/issues/960
```
Comment (meiersi-da) [inline ledger-api/grpc-definitions/com/daml/ledger/api/v1/event.proto:31]: Why keep the comment here?

Comment (meiersi-da) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/event.proto:31]: I'd suggest to add commentary on the field referencing #960 and explaining why it was re-introduced.

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/event.proto [hunk 3]
```diff
@@ -25,11 +25,12 @@ option csharp_namespace = "Com.Daml.Ledger.Api.V1";
 message Event {
   oneof event {
     CreatedEvent created = 1;
+    ExercisedEvent exercised = 2;
```
Comment (meiersi-da) [inline ledger-api/grpc-definitions/com/daml/ledger/api/v1/event.proto:28]: I'd suggest adding commentary explaining what kind of `ExercisedEvent` values can be found here, and when they will be present.

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/transaction_service.proto [hunk 4]
```diff
@@ -115,6 +115,9 @@ message GetTransactionsRequest {
   // Optional
   TraceContext trace_context = 1000;
 
+  // Optional
+  bool include_non_consuming = 6;
```
Comment (meiersi-da) [inline ledger-api/grpc-definitions/com/daml/ledger/api/v1/transaction_service.proto:119]: Add commentary on what this flag does.

Comment (meiersi-da) [inline ledger-api/grpc-definitions/com/daml/ledger/api/v1/transaction_service.proto:119]:
```suggestion
  bool include_non_consuming_exercise_events = 6;
```

Comment (meiersi-da) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/transaction_service.proto:119]: Or just change the name to be very explicit; and thereby leaving ample space for later additions of flags.

**Review by meiersi-da — COMMENTED** [review]
@mzagorski-da : thanks for the PR. This goes in the right direction. I believe it would be good to make this PR to change the protocol as final as possible, so we have clarity on the API change up front.

We can then switch to writing the design doc and reference the PR from there.

**Comment by kamil-da** [comment]
There is another PR #8876 which is a direct continuation of this draft. Closing.

**Pull request closed without merging**